### PR TITLE
Update AWS CloudFormation template to include AWS private API Gateway

### PR DESCRIPTION
**Pull request by sfc-gh-ahu**

### CODE REVIEW

File: DeploymentTemplates/aws/BasicSetup.yaml [hunk 1]
```diff
@@ -30,6 +38,14 @@ Parameters:
     AllowedPattern: "^[a-zA-Z0-9]+[-a-zA-Z0-9-]+[-a-zA-Z0-9]+$"
     Description: "Role used by the Lambda."
 
+  sourceVpcId:
+    Type: "String"
+    Default: ""
```
Comment (sf-gh-schintala) [inline DeploymentTemplates/aws/BasicSetup.yaml:43]: Would it be possible to ask for the VPC id, only if we select Private endpoints as apiGatewayType? This field shows up even when I pick Region endpoints as the apiGatewayType

sourceVpcId
Snowflake VPC that has access to private API Gateway. Used only when creating a private API Gateway

Comment (sfc-gh-ahu) [reply DeploymentTemplates/aws/BasicSetup.yaml:43]: i looked online, but it doesn't seem like cloudformation supports optional parameters :(

Comment (sf-gh-schintala) [reply DeploymentTemplates/aws/BasicSetup.yaml:43]: gotcha. no worries. would have been nice, but should be fine.

**Review by sf-gh-schintala — APPROVED** [review]
was able to test the CFT for setting up private endpoints and everything works as expected..

**Pull request merged**